### PR TITLE
Start using functions of kube-thanos

### PR DESCRIPTION
**Pull request by metalmatze**
Refactoring to use functions that kube-thanos exposes going forward.

### CODE REVIEW

File: environments/base/manifests/thanos-receive-default-service.yaml [hunk 1]
```diff
@@ -15,13 +15,13 @@ spec:
   ports:
   - name: grpc
     port: 10901
-    targetPort: 10901
+    targetPort: grpc
```
Comment (kakkoyun) [inline environments/base/manifests/thanos-receive-default-service.yaml:18]: Let's be consistent with target ports and use the actual port number not the name or vice versa.

File: environments/base/manifests/thanos-store-shard0-service.yaml [hunk 2]
```diff
@@ -15,10 +15,10 @@ spec:
   ports:
   - name: grpc
     port: 10901
-    targetPort: 10901
+    targetPort: grpc
```
Comment (kakkoyun) [inline environments/base/manifests/thanos-store-shard0-service.yaml:18]: ditto

File: environments/dev/manifests/thanos-receive-default-service.yaml [hunk 3]
```diff
@@ -15,13 +15,13 @@ spec:
   ports:
   - name: grpc
     port: 10901
-    targetPort: 10901
+    targetPort: grpc
```
Comment (kakkoyun) [inline environments/dev/manifests/thanos-receive-default-service.yaml:18]: ditto

File: environments/dev/manifests/thanos-store-shard0-service.yaml [hunk 4]
```diff
@@ -15,10 +15,10 @@ spec:
   ports:
   - name: grpc
     port: 10901
-    targetPort: 10901
+    targetPort: grpc
```
Comment (kakkoyun) [inline environments/dev/manifests/thanos-store-shard0-service.yaml:18]: ditto

**Review by kakkoyun — COMMENTED** [review]
Looks great.
My only comment concerning ports, we should be consistent and use only name or port number for target ports.

File: environments/base/manifests/thanos-receive-default-service.yaml (hunk 1 above)
Comment (metalmatze) [reply environments/base/manifests/thanos-receive-default-service.yaml:18]: That's actually a kube-thanos change and I made this after looking at the Kubernetes docs on Services.
I guess we want the port _number_ for both port and targetPort then?

Comment (kakkoyun) [reply environments/base/manifests/thanos-receive-default-service.yaml:18]: Yes, I believe it's simpler and straightforward. However, if it's the _recommended_ way of doing it then let's use it everywhere.

Comment (metalmatze) [reply environments/base/manifests/thanos-receive-default-service.yaml:18]: It's not very clear and I feel like there's no definite answer. However, now I see that the docs say:

> Note: A Service can map any incoming port to a targetPort. By default and for convenience, the targetPort is set to the same value as the port field.

I'll move to `port == targetPort` then :man_shrugging: :smile:

**Review by kakkoyun — APPROVED** [review]
🥇 LGTM

**Comment by metalmatze** [comment]
nice

**Pull request merged**